earth2studio/data/arco.py

```diff
@@ -260,18 +253,42 @@ def fetch_array(self, time: datetime, variable: str) -> np.ndarray:
             raise e
 
         arco_variable, level = arco_name.split("::")
-
-        shape = self.zarr_group[arco_variable].shape
-        # Static variables
-        if len(shape) == 2:
-            output = modifier(self.zarr_group[arco_variable][:])
-        # Surface variable
-        elif len(shape) == 3:
-            output = modifier(self.zarr_group[arco_variable][time_index])
-        # Atmospheric variable
+        if self.zarr_major_version >= 3:
+            zarr_array = await self.zarr_group.get(arco_variable)
+            shape = zarr_array.shape
+            # Static variables
+            if len(shape) == 2:
+                data = await zarr_array.getitem(slice(None))
+                output = modifier(data)
+            # Surface variable
+            elif len(shape) == 3:
+                data = await zarr_array.getitem(time_index)
+                output = modifier(data)
+            # Atmospheric variable
+            else:
+                # Load levels coordinate system from Zarr store and check
+                level_coords = await (await self.zarr_group.get("level")).getitem(
```

Comment:
Can you not cache this in the constructor so each call doesn't have to do this? Also, each variable will do this so caching will also eliminate redundancy here?

Reply:
Good point, move it up out of this function